pkg/sqlcache/informer/listoption_indexer_test.go

```diff
@@ -264,40 +297,112 @@ func TestNewListOptionIndexer(t *testing.T) {
 	}
 }
 
-func TestListByOptions(t *testing.T) {
+func TestNewListOptionIndexerEasy(t *testing.T) {
+	ctx := context.Background()
+
 	type testCase struct {
-		description           string
-		listOptions           sqltypes.ListOptions
-		partitions            []partition.Partition
-		ns                    string
-		expectedCountStmt     string
-		expectedCountStmtArgs []any
-		expectedStmt          string
-		expectedStmtArgs      []any
-		extraIndexedFields    []string
-		expectedList          *unstructured.UnstructuredList
-		returnList            []any
-		expectedContToken     string
-		expectedErr           error
+		description string
+		listOptions sqltypes.ListOptions
+		partitions  []partition.Partition
+		ns          string
+
+		items []*unstructured.Unstructured
+
+		extraIndexedFields [][]string
+		expectedList       *unstructured.UnstructuredList
+		expectedTotal      int
+		expectedContToken  string
+		expectedErr        error
+	}
+	foo := map[string]any{
+		"metadata": map[string]any{
+			"name":      "obj1",
+			"namespace": "ns-a",
+			"somefield": "foo",
+			"sortfield": "4",
+		},
+	}
+	bar := map[string]any{
+		"metadata": map[string]any{
+			"name":      "obj2",
+			"namespace": "ns-a",
+			"somefield": "bar",
+			"sortfield": "1",
+			"labels": map[string]any{
+				"cows":   "milk",
+				"horses": "saddles",
+			},
+		},
 	}
+	baz := map[string]any{
+		"metadata": map[string]any{
+			"name":      "obj3",
+			"namespace": "ns-a",
+			"somefield": "baz",
+			"sortfield": "2",
+			"labels": map[string]any{
+				"horses": "saddles",
+			},
+		},
+		"status": map[string]any{
+			"someotherfield": "helloworld",
+		},
+	}
+	toto := map[string]any{
+		"metadata": map[string]any{
+			"name":      "obj4",
+			"namespace": "ns-a",
+			"somefield": "toto",
+			"sortfield": "2",
+			"labels": map[string]any{
+				"cows": "milk",
+			},
+		},
+	}
+	lodgePole := map[string]any{
+		"metadata": map[string]any{
+			"name":      "obj5",
+			"namespace": "ns-b",
+			"unknown":   "hi",
+			"labels": map[string]any{
+				"guard.cattle.io": "lodgepole",
+			},
+		},
+	}
+
+	makeList := func(t *testing.T, objs ...map[string]any) *unstructured.UnstructuredList {
```

Comment:
I would prefer this function to be pulled out and made package-level because it doesn't appear to be a closure on any variables in the enclosing scope. I realize it's only used in this test, but it's useful and we might want to call it from other tests.